### PR TITLE
HPCC-17281 Support reading/writing fields of type BIT(nn) in MySQL

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
Updated regression suite test to cover.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

**Comment by richardkchapman** [comment]
@AttilaVamos Any idea what the failure was?

File: testing/regress/ecl/mysqlembed.ecl [hunk 1]
```diff
@@ -47,27 +49,27 @@ stringrec extractName(childrec l) := TRANSFORM
...
+init := DATASET([{'name1', 0x4161, 1, true, 1.2, 3.4, D'aa55aa55', 1234567.89, U'Straße', U'Straße'},
+                 {'name2', 66, 2, false, 5.6, 7.8, D'00', -1234567.89, U'là', U'là', '2015-12-25 01:23:45' }], childrec);
 
 drop() := EMBED(mysql : server(myServer),user(myUser),database(myDB))
   DROP TABLE IF EXISTS tbl1;
 ENDEMBED;
 
 create() := EMBED(mysql : server(myServer),user(myUser),database(myDB))
-  CREATE TABLE tbl1 ( name VARCHAR(20), value INT, boolval TINYINT, r8 DOUBLE, r4 FLOAT, d BLOB, ddd DECIMAL(10,2), u1 VARCHAR(10), u2 VARCHAR(10), dt DATETIME );
+  CREATE TABLE tbl1 ( name VARCHAR(20), bval BIT(15), value INT, boolval TINYINT, r8 DOUBLE, r4 FLOAT, d BLOB, ddd DECIMAL(10,2), u1 VARCHAR(10), u2 VARCHAR(10), dt DATETIME );
```
Comment (ghalliday) [inline testing/regress/ecl/mysqlembed.ecl]: Here bitfield is before the value, but in the ecl record it comes after it.  Was this intended?

**Comment by AttilaVamos** [comment]
I'm checking

**Review by ghalliday — COMMENTED** [review]
@richardkchapman one question on the test

File: testing/regress/ecl/mysqlembed.ecl (hunk 1 above)
Comment (richardkchapman) [inline testing/regress/ecl/mysqlembed.ecl]: It was not (though it probably doesn't make any difference!). I will fix

**Comment by AttilaVamos** [comment]
@richardkchapman The problem is with MySQL server:
```
Failed to connect (Access denied for user &apos;rchapman&apos;@&apos;localhost&apos; (using password: NO))
```
I'm working on the fix.

**Comment by AttilaVamos** [comment]
@richardkchapman MySQL server is OK, now.

File: testing/regress/ecl/mysqlembed.ecl (hunk 1 above)
Comment (richardkchapman) [inline testing/regress/ecl/mysqlembed.ecl]: Fixed and repushed

**Comment by richardkchapman** [comment]
@ghalliday I have fixed the typo in the test program

**Pull request merged**